components/brave_sync/brave_profile_sync_service_impl.cc

```diff
@@ -823,7 +823,28 @@ void BraveProfileSyncServiceImpl::FetchSyncRecords(const bool bookmarks,
     category_names.push_back(kPreferences);  // "PREFERENCES";
   }
 
-  base::Time start_at_time = brave_sync_prefs_->GetLatestRecordTime();
+  base::Time start_at_time;
+  // Similar to FetchDevices
+  // During 70 sec after device connected to chain use start_at parameter
+  // of empty to force fetch from S3.
+  // We need this to handle the case:
+  // 1) deviceB connected to chain, created own queues
+  // 2) deviceB made the first fetch from S3, got the records and set own
+  //    non-empty latest_bookmark_record_time, so the next fetches would
+  //    be done through SQS
+  // 3) deviceA sends record
+  // 4) lambda enumerates queues and could not discover queues from deviceB,
+  //    because there is gap ~10~30 sec
+  // 5) record does not arrive to queue of deviceB and is lost for deviceB
+  // Any possibility of duplication will be eliminated by alreadySeenFromS3
+  // checks in brave_sync/extension/brave-sync/lib/s3Helper.js.
+  if (!tools::IsTimeEmpty(this_device_created_time_) &&
```

Comment:
can we pull out the check into a function named like `IsSQSReady` and merge the comments because same logic is duplicate in two places

Reply:
Addressed and made forced push https://github.com/brave/brave-core/pull/4168/files#diff-ab344835b09fb6ac1c601af787b116bbR797 .